File: edukate-common/src/main/java/io/github/sanyavertolet/edukate/common/users/UserCredentials.java
package io.github.sanyavertolet.edukate.common.users;

import lombok.Builder;
import lombok.Data;

import java.util.Set;

@Data
@Builder
public class UserCredentials {
    private final String id;
    private final String username;
    private final String encodedPassword;
    private final String email;
    private final Set<UserRole> roles;
    private final UserStatus status;

    public static UserCredentials newUser(String username, String encodedPassword, String email) {
        return UserCredentials.builder()
                .id(null)
                .username(username)
                .encodedPassword(encodedPassword)
                .email(email)
                .roles(Set.of(UserRole.USER))
                .status(UserStatus.PENDING)
                .build();
    }
}

File: edukate-common/src/main/kotlin/io/github/sanyavertolet/edukate/common/users/UserCredentials.kt
package io.github.sanyavertolet.edukate.common.users

data class UserCredentials(
    val id: Long?,
    val username: String,
    val encodedPassword: String,
    val email: String? = null,
    val roles: Set<UserRole>,
    val status: UserStatus,
) {
    override fun toString(): String =
        "UserCredentials(id=$id, username='$username', email='$email', roles=$roles, status=$status)"

    companion object {
        @JvmStatic
        fun newUser(username: String, encodedPassword: String, email: String) =
            UserCredentials(null, username, encodedPassword, email, setOf(UserRole.USER), UserStatus.PENDING)
    }
}
